### ISSUE TITLE
self defined gradient for certain value of input

### ISSUE THREAD

**Issue by zyzhang1130** [HIPS/autograd#573]
Is it possible to self define gradient value for certain input value where the gradient is not defined? One example would be x= 0 for f(x) = 1/x

**Comment by CamDavidsonPilon** [comment]
Yes, but you'll need to dip into the internals of autograd library. Here's an example of creating auto-diff functions compatible with autograd: https://github.com/CamDavidsonPilon/autograd-gamma/blob/master/autograd_gamma/__init__.py

So it's possible, but requires more work.
